lib/createGltf.js

```diff
@@ -1,347 +1,394 @@
+    var i;
+    var material;
+    var materialsLength = materials.length;
+    for (i = 0; i < materialsLength; ++i) {
+        if (materials[i].name === materialName) {
+            material = materials[i];
+            break;
+        }
     }
 
-    function requiresUint32Indices(nodes) {
-        var nodesLength = nodes.length;
-        for (var i = 0; i < nodesLength; ++i) {
-            var meshes = nodes[i].meshes;
-            var meshesLength = meshes.length;
-            for (var j = 0; j < meshesLength; ++j) {
-                // Reserve the 65535 index for primitive restart
-                var vertexCount = meshes[j].positions.length / 3;
-                if (vertexCount > 65534) {
-                    return true;
-                }
-            }
+    if (!defined(material)) {
+        material = getDefaultMaterial(options);
+        material.name = materialName;
+    }
+
+    var materialIndex;
+    materialsLength = gltf.materials.length;
+    for (i = 0; i < materialsLength; ++i) {
+        if (gltf.materials[i].name === materialName) {
+            materialIndex = i;
+            break;
         }
-        return false;
     }
 
-    var uint32Indices = requiresUint32Indices(nodes);
-    var gltfSceneNodes = gltf.scenes[sceneId].nodes;
+    if (!defined(materialIndex)) {
+        materialIndex = addMaterial(gltf, material);
+    }
+
+    return materialIndex;
+}
+
+function addVertexAttribute(gltf, array, components, name) {
+    var count = array.length / components;
+    var minMax = array.getMinMax(components);
+    var type = (components === 3 ? 'VEC3' : 'VEC2');
+
+    var accessor = {
+        name : name,
+        componentType : WebGLConstants.FLOAT,
+        count : count,
+        min : minMax.min,
+        max : minMax.max,
+        type : type
+    };
+
+    var accessorIndex = gltf.accessors.length;
+    gltf.accessors.push(accessor);
+    return accessorIndex;
+}
+
+function addIndexArray(gltf, array, uint32Indices, name) {
+    var componentType = uint32Indices ? WebGLConstants.UNSIGNED_INT : WebGLConstants.UNSIGNED_SHORT;
+    var count = array.length;
+    var minMax = array.getMinMax(1);
+
+    var accessor = {
+        name : name,
+        componentType : componentType,
+        count : count,
+        min : minMax.min,
+        max : minMax.max,
+        type : 'SCALAR'
+    };
+
+    var accessorIndex = gltf.accessors.length;
+    gltf.accessors.push(accessor);
+    return accessorIndex;
+}
+
+function requiresUint32Indices(nodes) {
     var nodesLength = nodes.length;
     for (var i = 0; i < nodesLength; ++i) {
-        // Add node
-        var node = nodes[i];
-        var nodeId = node.name;
-        gltfSceneNodes.push(nodeId);
-        var gltfNodeMeshes = [];
-        gltf.nodes[nodeId] = {
-            name : nodeId,
-            meshes : gltfNodeMeshes
-        };
-
-        // Add meshes to node
-        var meshes = node.meshes;
+        var meshes = nodes[i].meshes;
```

Comment:
That's what I used to do before I realized Cesium does not like different index types being in the same buffer view. I think the situation has changed a bit since different sized indices wouldn't be allowed in the same bufferView anyways in gltf 2.0.

I do like the simplicity of this approach though, but I'll open an issue since I can think of some models that would require more memory because of this.